% add section on what phages do in the environemnt, in general. Dont constrain to applications of the phages, talk about what they do, in the larger environemnt ecosystem, carbon turnover, 30% (depends on ecosystem) of bacterial lysis per day. (done)


% Read review from matti

% add section on thesis overview, how I created the tools and run the visualizations/analysis (done)

% In the different models (2.1), mention how phages and resources can be added to the model itself. 
% flip it around to say how you build form modelling just bacteria, maybe resources, to adding resources and phages. 
% add section 2.1.5 on a model that specifically models phages. 
% tell how the Geng model cna be made form the other 2.1.1-4 models. 

% Repression of DNA be more general, move outside and mention that you cna go to lytic or lysogenic part

% holin to lyse bacteria (done)

% do I really need fitness vlaue of the cell, and horizontal gene transfer? Maybe a bit too much, cut it down, or just rmeove it? (done, by cutting parts of it)

% CHange “good plot” to something like “this is waht typically looks like in a lab”. (done)




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Masters/Doctoral Thesis 
% LaTeX Template
% Version 1.43 (17/5/14)
%
% This template has been downloaded from:
% http://www.LaTeXTemplates.com
%
% Original authors:
% Steven Gunn 
% http://users.ecs.soton.ac.uk/srg/softwaretools/document/templates/
% and
% Sunil Patel
% http://www.sunilpatel.co.uk/thesis-template/
%
% License:
% CC BY-NC-SA 3.0 (http://creativecommons.org/licenses/by-nc-sa/3.0/)
%
% Note:
% Make sure to edit document variables in the Thesis.cls file
%
% Modified and Adapted by Michael Lees 2020
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%	PACKAGES AND OTHER DOCUMENT CONFIGURATIONS
%----------------------------------------------------------------------------------------

\documentclass[11pt, oneside, draft]{Thesis}
% \documentclass[11pt, oneside, final]{Thesis}

\graphicspath{{Images/}} 
\usepackage[square, numbers, comma, sort&compress]{natbib} % Use the natbib reference package - read up on this to edit the reference style; if you want text (e.g. Smith et al., 2012) for the in-text references (instead of numbers), remove 'numbers' 
\usepackage[ruled,vlined]{algorithm2e}
\usepackage{amsmath}
\usepackage{booktabs}
\usepackage[nameinlink]{cleveref}
\PassOptionsToPackage{hyphens}{url}
\usepackage{hyperref}
\usepackage{array}
\usepackage[svgnames]{xcolor}
\usepackage{textcomp}
\usepackage[font=small,labelfont=bf]{caption}
\usepackage{float}
\usepackage{subcaption}
\usepackage{tabularx}
% \usepackage[margin=1in]{geometry} % Adjust margins if needed
\usepackage{booktabs}    
\usepackage{listings}
\usepackage{bm}
\usepackage{aliascnt}
\usepackage{algpseudocode}
\usepackage{makecell}
\floatstyle{ruled}
\newfloat{listing}{htbp}{lop}
\floatname{listing}{Algorithm}
\newsavebox{\codebox}
\newaliascnt{eqfloat}{equation}
\newfloat{eqfloat}{h}{eqflts}
\floatname{eqfloat}{Equation}
\newcommand*{\ORGeqfloat}{}
\let\ORGeqfloat\eqfloat
\def\eqfloat{%
  \let\ORIGINALcaption\caption
  \def\caption{%
    \addtocounter{equation}{-1}%
    \ORIGINALcaption
  }
  \ORGeqfloat
}
\DeclareMathAlphabet{\mathcal}{OMS}{cmsy}{m}{n}
\SetMathAlphabet{\mathcal}{bold}{OMS}{cmsy}{b}{n}

% styling used for code color in appendix D
\definecolor{codegreen}{rgb}{0,0.6,0}
\definecolor{codegray}{rgb}{0.5,0.5,0.5}
\definecolor{codepurple}{rgb}{0.58,0,0.82}
\definecolor{backcolour}{rgb}{0.95,0.95,0.92}
\lstdefinestyle{codestyle}{
    backgroundcolor=\color{backcolour}, 
    commentstyle=\color{codegreen}, 
    keywordstyle=\color{magenta}, 
    numberstyle=\tiny\color{codegray}, 
    stringstyle=\color{codepurple}, 
    basicstyle=\ttfamily\footnotesize, 
    breakatwhitespace=false, 
    breaklines=true, 
    captionpos=b, 
    language=Python, 
    keepspaces=true, 
    numbers=left, 
    numbersep=10pt, 
    showtabs=false, 
    tabsize=2 
}

% For better horizontal rules
\DeclareMathOperator*{\argmin}{arg\,min} 
\DeclareMathOperator*{\argmax}{arg\,max}
\DeclareCaptionType{equ}[][]
\hypersetup{colorlinks=true, filecolor=magenta, urlcolor=cyan, linkcolor=blue} % Colors hyperlinks in blue - change to black if annoying
\urlstyle{same}
\title{\ttitle} % Defines the thesis title - don't touch this
\begin{document}
\frontmatter 
\setstretch{1.3} 

\fancyhead{} 
\rhead{\thepage} 
\lhead{}

\pagestyle{fancy} % Finally, use the "fancy" page style to implement the FancyHdr headers

\newcommand{\HRule}{\rule{\linewidth}{0.5mm}} 

\hypersetup{pdftitle={\ttitle}}
\hypersetup{pdfsubject=\subjectname}
\hypersetup{pdfauthor=\authornames}
\hypersetup{pdfkeywords=\keywordnames}

%----------------------------------------------------------------------------------------
%	TITLE PAGE
%----------------------------------------------------------------------------------------

\begin{titlepage}
\begin{center}

\textsc{\LARGE \univname}\\[1.5cm] % University name
\textsc{\Large Masters Thesis}\\[0.5cm] % Thesis type

\HRule \\[0.4cm] % Horizontal line
{\huge \bfseries \ttitle}\\[0.4cm] % Thesis title
\HRule \\[1.5cm] % Horizontal line
 
\begin{minipage}{0.4\textwidth}
\begin{flushleft} \large
\emph{Author:}\\
\authornames % Author name - remove the \href bracket to remove the link
\end{flushleft}
\end{minipage}
\begin{minipage}{0.4\textwidth}
\begin{flushright} \large
\emph{Examiner:} \\
{\exname}\\
\emph{Supervisor:} \\
{\supname}\\
\emph{Assessor:} \\
{\assessorname}
\end{flushright}
\end{minipage}\\[1cm]
 
\large \textit{A thesis submitted in partial fulfillment of the requirements\\ for the degree of \degreename}\\[0.3cm] % University requirement text
\textit{in the}\\[0.4cm]
\groupname\\\deptname\\[1cm] % Research group name and department name
 
{\large \todayy}\\[2cm] % Date
\includegraphics[width=0.6\textwidth]{clslogo.png} % Include Computational Science Logo
 
\vfill
\end{center}

\end{titlepage}

%----------------------------------------------------------------------------------------
%	DECLARATION PAGE
%	Your institution may give you a different text to place here
%----------------------------------------------------------------------------------------

\Declaration{

\addtocontents{toc}{\vspace{1em}} % Add a gap in the Contents, for aesthetics

I, \authornames, declare that this thesis, entitled `\ttitle' and the work presented in it are my own. I confirm that:

\begin{itemize} 
\item[\tiny{$\blacksquare$}] This work was done wholly or mainly while in candidature for a research degree at the University of Amsterdam.
\item[\tiny{$\blacksquare$}] Where any part of this thesis has previously been submitted for a degree or any other qualification at this University or any other institution, this has been clearly stated.
\item[\tiny{$\blacksquare$}] Where I have consulted the published work of others, this is always clearly attributed.
\item[\tiny{$\blacksquare$}] Where I have quoted from the work of others, the source is always given. With the exception of such quotations, this thesis is entirely my own work.
\item[\tiny{$\blacksquare$}] I have acknowledged all main sources of help.
\item[\tiny{$\blacksquare$}] Where the thesis is based on work done by myself jointly with others, I have made clear exactly what was done by others and what I have contributed myself.\\
\end{itemize}


Signed: \\\\\includegraphics[width=3cm]{Images/signature.pdf} \\
%\rule[1em]{25em}{0.5pt} % This prints a line for the signature
 
Date: \today\\
%\rule[1em]{25em}{0.5pt} % This prints a line to write the date
}

\clearpage % Start a new page

%----------------------------------------------------------------------------------------
%	QUOTATION PAGE
%----------------------------------------------------------------------------------------

\pagestyle{empty}
\null\vfill 
\textit{``All models are wrong, but some are useful``}
\begin{flushright}
    George E. P. Box
\end{flushright}
\vfill\vfill\vfill\vfill\vfill\vfill\null 
\clearpage 

%----------------------------------------------------------------------------------------
%	ABSTRACT PAGE
%----------------------------------------------------------------------------------------
\addtotoc{Abstract} 
\abstract{\addtocontents{toc}{\vspace{1em}} 
    In any given microbial ecosystem, there can be hundreds of different phages and bacteria interacting with one another in a complex system. 
    These individual interactions have a significant impact on the bacterial community and the overall ecosystem, with important ecological consequences. 
They force bacteria to mutate and evolve, changing the community fitness level. 
    Phages can control bacterial populations, ensuring that dangerous bacteria do not over-reproduce and harm the ecosystem. 
    Phages control bacterial population dynamics using the “kill-the-winner” dynamic, preventing any one species from dominating.
    Upon lysis, bacteria release resources into the environment that other bacteria and plants can use. 

    There is a need to mathematically model complex systems, but current models typically only model 1 or 2 phages and bacteria. 
    It is, however, straightforward to extend the model to multiple species. 
    Mathematically modeling large bacterial communities is important because running lab experiments takes time and is costly. 
    These simulations can guide future lab work, for example, in combating antibiotic-resistant bacteria. 

    I created a program consisting of three parts that can model complex communities of $p$ phages, $b$ bacteria, and $r$ resources. 
    The first part uses a GUI tool to create and edit the interactions between the phages, bacteria, and resources using a graph network. 
    The nodes and edges hold the interaction and environmental parameter values. 
    The second part, the simulation framework itself, models and calculates the phage, bacteria, and resource population levels with the user-provided network and ODE model. 
    The third part allows the user to interact with the simulation framework with a custom-built dashboard. 
    The user can, for example, change the parameter values from the dashboard and immediately see how the change affects the simulation. 
    There are five pre-built plots that the user can interact with. 
    It is possible to download the simulation results so that the user can create custom analyses and plots. 

    I apply this process to the Golding model from \citet{gengUsingBacterialPopulation2024}. 
    I qualitatively analyze the model's behavior and use a SOBOL analysis to quantitatively assign a value to the model's variance in output from the model's input. 
    In addition, understanding the different regimes of restriction in phages, as well as the factors that drive reaction rates and determine the rate-limiting steps, is crucial.
    By varying the initial uninfected bacteria population, the model can interpolate between two limiting scenarios, adsorption and latency limited. 
    It is important to determine whether a phage population will proliferate in a laboratory context under constant washout to prevent the phage population from accidentally being wiped out. 
    I run an analysis to see if phages will proliferate or die out depending on the initial condition. 
    Finally, I perform and analyze a co-existence analysis on a large 20-phage, 20-bacteria, and 10-resource community to determine whether phages and bacteria can coexist under various conditions. 
    I find that in large communities, phages and bacteria can coexist. 
    If a debris term is introduced, the bacteria's survivability rate increases.

% Include your abstract here 
% Abstracts must include sufficient information for reviewers to judge the nature and significance
% of the topic, the adequacy of the investigative strategy, the nature of the results, and the
% conclusions. The abstract should summarize the substantive results of the work and not merely
% list topics to be discussed. 
% Length 200-400 words.
}
\clearpage 

%----------------------------------------------------------------------------------------
%	ACKNOWLEDGEMENTS
%----------------------------------------------------------------------------------------

\setstretch{1.3} 
\acknowledgements{
    \addtocontents{toc}{\vspace{1em}} 
    I would like to thank my parents for loving me despite some of my faults, and for supporting me through my Bachelor and Master studies even if they don't exactly know what I am studying. Without them, I wouldn't know where my life would be right now, and I certainly would be a different person if it were not for them. 

    Thank you to Dr. Matti Gralka for the weekly meetings and teaching me everything about phages and bacteria. 
    Every meeting was always insightful, productive, and informative. 
    I will forever be amazed at how he can remember which paper talks about which topic, and how he always had a paper for every topic. 

    Thank you to Sofia Blaszczyk for finding the Master thesis opening and suggesting that I email Dr. Gralka for an introductory meeting. 
    She acted as my \href{https://www.wikiwand.com/en/articles/Rubber_duck_debugging}{rubber duck programming buddy}, and watched my cringe screen recordings that I sent her at 2am showcasing various demos of my project. 
    If she wouldn't have found this opening, I wouldn't know what I would be doing as my thesis. 

    If I hadn't followed Dr. Rik Kaasschieter's and Dr. Martijn Anthonissen's courses “Introduction Computational Sciences“ and “Numerical Linear Algebra“ in my Bachelors, I would not have been interested in Computational Sciences. 
    I would not have found the MSc Computational Sciences program, as Computational Sciences fits my interests and skill sets better than any other program I could have taken. 
    For Rik and Martijn have forever altered my career trajectory. 

    Thank you to Sarah Flickinger for showing me the research that she has been doing in the lab. 
    She allowed me to really connect my research and models to real life, reminding me that what I am doing has real life use cases than just a purely theoretical or programming challenge. 

    And finally, thank you to all of my friends for keeping me sane and helping me through both of my programs. 
}
\clearpage 

%----------------------------------------------------------------------------------------
%	LIST OF CONTENTS/FIGURES/TABLES PAGES
%----------------------------------------------------------------------------------------

\pagestyle{fancy} 
\lhead{\emph{Contents}} 
\tableofcontents 
\lhead{}

\lhead{\emph{List of Figures}} 
\listoffigures 
\lhead{}

\lhead{\emph{List of Tables}}
\listoftables
\lhead{}

% \lhead{\emph{List of Algorithms}}
% \addtotoc{List of Algorithms}
% \listofalgorithms
% \lhead{}

%----------------------------------------------------------------------------------------
%	ABBREVIATIONS
%----------------------------------------------------------------------------------------

\clearpage 
\setstretch{1.5}
\lhead{\emph{Abbreviations}}
\listofsymbols{ll}
{
    % \textbf{ABM} & \textbf{A}gent \textbf{B}ased \textbf{M}odelling \\ 
    % \textbf{ARD} & \textbf{A}rms \textbf{R}ace \textbf{D}ynamic \\ 
    % \textbf{BVP} & \textbf{B}oundary \textbf{V}alue \textbf{P}roblem \\ 
    % \textbf{CBASS} & \textbf{C}yclic oligonucleotide-\textbf{B}ased \textbf{A}ntiphage \textbf{S}ignalling \textbf{S}ystems\\
    % \textbf{CRISPR} & \textbf{C}lustered \textbf{R}egularly \textbf{I}nterspaced \textbf{S}hort \textbf{P}alindromic \textbf{R}epeats \\
    \textbf{DDE} & \textbf{D}elay \textbf{D}ifferential \textbf{E}quation \\ 
    \textbf{DNA} & \textbf{D}eoxyribo\textbf{N}ucleic \textbf{A}cid \\
    % \textbf{FSD} & \textbf{F}luctuating \textbf{S}election \textbf{D}ynamics \\ 
    \textbf{GUI} & \textbf{G}raphical \textbf{U}ser \textbf{I}nterface \\ 
    \textbf{IC} & \textbf{I}nitial \textbf{C}ondition \\
    \textbf{IVA} & \textbf{I}nitial \textbf{V}alue \textbf{A}nalysis \\ 
    \textbf{OD} & \textbf{O}ptical \textbf{D}ensity \\ 
    \textbf{ODE} & \textbf{O}rdinary \textbf{D}ifferential \textbf{E}quation \\
    \textbf{PA} & \textbf{P}arameter \textbf{A}nalysis \\
    \textbf{PDE} & \textbf{P}artial \textbf{D}ifferential \textbf{E}quation \\ 
    \textbf{RNA} & \textbf{R}ibo\textbf{N}ucleic \textbf{A}cid\\
    % \textbf{SIE} & \textbf{S}uper\textbf{I}nfection \textbf{E}xclusion \\
    % \textbf{SNP} & \textbf{S}ingle \textbf{N}ucleotide \textbf{P}olymorphism \\
    \textbf{ST} & \textbf{S}erial \textbf{T}ransfer\\ 
    % \textbf{TAB} & \textbf{T}ail \textbf{A}ssembly \textbf{B}locker \\
    \textbf{UA} & \textbf{U}ltimate \textbf{A}nalysis \\
    \textbf{UvA} & \textbf{U}niversitiet \textbf{v}an \textbf{A}msterdam\\
}
\lhead{}

%----------------------------------------------------------------------------------------
%	PHYSICAL CONSTANTS/OTHER DEFINITIONS
%----------------------------------------------------------------------------------------

\clearpage 

%\lhead{\emph{Physical Constants}} % Set the left side page header to "Physical Constants"

%\listofconstants{lrcl} % Include a list of Physical Constants (a four column table)
%{
%Speed of Light & $c$ & $=$ & $2.997\ 924\ 58\times10^{8}\ \mbox{ms}^{-\mbox{s}}$ (exact)\\
%% Constant Name & Symbol & = & Constant Value (with units) \\
%}

%----------------------------------------------------------------------------------------
%	SYMBOLS
%----------------------------------------------------------------------------------------

\clearpage

% \lhead{\emph{Symbols}} 

% A table of Symbols and Parameter Values can be viewed in \Cref{AppendixA} and \Cref{AppendixE}. 
% \listofnomenclature{lll} % Include a list of Symbols (a three column table)
% {
% $a$ & distance & m \\
% $P$ & power & W (Js$^{-1}$) \\
% % Symbol & Name & Unit \\

% & & \\ % Gap to separate the Roman symbols from the Greek

% $\omega$ & angular frequency & rads$^{-1}$ \\
% % Symbol & Name & Unit \\
% }

%----------------------------------------------------------------------------------------
%	DEDICATION
%----------------------------------------------------------------------------------------

\setstretch{1.3} 
\pagestyle{empty} 
\addtocontents{toc}{\vspace{2em}} 

%----------------------------------------------------------------------------------------
%	THESIS CONTENT - CHAPTERS
%----------------------------------------------------------------------------------------

\mainmatter
\pagestyle{fancy}

%This structure provides a bare bones essentials of the thesis and some indicative length
%This structure may not fit your thesis perfectly, but be sure to include these components somehow.
%It is possible to split the chapters up (E.g., 2 methods chapter, Experiment and Results as two, etc.)

% The typical length may be between 43 - 64 pages. Do not worry if you go slightly larger or smaller than this. 
% But a thesis of 20 pages, or 100 pages may suggest you've been to brief or too verbose.
%NOTE - there is not strict limit/minimum.

\lhead{\emph{Introduction}}
\input{Chapters/Introduction} %Set out your thesis and state your research question (5-8 pages)
\newpage

\lhead{\emph{Literature Review}}
\input{Chapters/LiteratureReview} %Related literature and material (8-12 pages)
\newpage

\lhead{\emph{Methods}}
\input{Chapters/Methods} %Details of your approach (10-15 pages)
\newpage

\lhead{\emph{Experiments and Results}}
\input{Chapters/ExperimentsandResults} %Evaluation/testing of your hypothesis (10-15 pages)
\newpage

\lhead{\emph{Discussion}}
\input{Chapters/Discussion} %Evaluation/testing of your hypothesis (5-8 pages)
\newpage

\lhead{\emph{Conclusions and Future Work}}
\input{Chapters/ConclusionsandFutureWork} % (5-6 pages)
\newpage

\lhead{\emph{Ethics and Datamanagemnt}}
\input{Chapters/EthicsandDatamanagemnt}
\newpage
\lhead{}

%----------------------------------------------------------------------------------------
%	THESIS CONTENT - APPENDICES
%----------------------------------------------------------------------------------------

\addtocontents{toc}{\vspace{2em}}
\appendix
\lhead{\emph{Appendix A}}
\input{Appendices/AppendixA}
\newpage

\lhead{\emph{Appendix B}}
\input{Appendices/AppendixB}
\newpage 

\lhead{\emph{Appendix C}}
\input{Appendices/AppendixC}
\newpage 

\lhead{\emph{Appendix D}}
\input{Appendices/AppendixD}
\newpage 

\lhead{\emph{Appendix E}}
\input{Appendices/AppendixE.tex}
\newpage 

\lhead{\emph{Appendix F}}
\input{Appendices/AppendixF.tex}
\newpage

\addtocontents{toc}{\vspace{2em}}
 
%----------------------------------------------------------------------------------------
%	BIBLIOGRAPHY
%----------------------------------------------------------------------------------------

\backmatter
\label{Bibliography}
\lhead{\emph{Bibliography}}
\bibliographystyle{unsrtnat}
\bibliography{References,References2} 
\end{document} 